### PR TITLE
add clap for args

### PR DESCRIPTION
**Pull request by byteZorvin**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -104,54 +104,66 @@ impl FileWriter {
...
 
-    for (_, path) in args[1..].iter().enumerate() {
+    #[arg(short, long)]
+    trace: String,
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+
+    for path in [
```
Comment (apoorvsadana) [inline src/main.rs:129]: it looks like you're using this to ensure all args were provided, i think you can use #[arg(required = true)]

Comment (byteZorvin) [reply src/main.rs:129]: No this is for making sure that the path exist in the os, and not random string

Comment (apoorvsadana) [reply src/main.rs:129]: I see, but we should probably still make the args required? Or is it not necessary?

Comment (byteZorvin) [reply src/main.rs:129]: Ouch yes, Made them required !!

**Pull request merged**